lib/recurly/relay.js

```diff
@@ -26,13 +26,15 @@ function relay (done) {
   }
 
   events.bind(window, 'message', function listener (event) {
-    var data = JSON.parse(event.data);
-    var name = data.recurly_event;
-    var body = data.recurly_message;
-    var err = body.error ? errors('api-error', body.error) : null;
-    events.unbind(window, 'message', listener);
-    if (name) self.emit(name, err, body);
-    if (frame) document.body.removeChild(frame);
+    if (event.origin === 'https://api.recurly.com') {
```

Comment:
Technically the intended origin can be any base domain provided by the `api` property passed to config. The property is available here as `this.config.api`, from which the base uri can be extracted.